src/worker/plugins/run.ts

```diff
@@ -27,6 +31,16 @@ export async function runPlugins(server: PluginsServer, event: PluginEvent): Pro
         }
     }
 
+    if (!returnedEvent.properties) {
+        returnedEvent.properties = {}
+    }
+    if (pluginsRan.length > 0) {
+        returnedEvent.properties['$plugins_ran_successfully'] = pluginsRan
+    }
+    if (pluginsFailed.length > 0) {
+        returnedEvent.properties['$plugins_failed'] = pluginsFailed
+    }
```

Comment:
Some code feedback then :). 

1) There's no need to run `if (!returnedEvent.properties) { returnedEvent.properties = {} }` if we're not going to change anything (when lengths are 0 for both arrays)
2) I think it's better that either both keys exist or none do, otherwise you'll end up second guessing yourself if the $plugins_failed array is just not there (instead of being empty, which tells you directly that nothing failed)

You could do something like:

```js
    if (pluginsRan.length > 0 || pluginsFailed.length > 0) {
        event.properties = { 
            ...event.properties,
            '$plugins_ran_successfully': pluginsRan,
            '$plugins_failed': pluginsFailed,
        }
    }
```